tests/integrations/pure_eval/test_pure_eval.py

```diff
@@ -28,8 +50,43 @@ def foo():
     frame_vars = event["exception"]["values"][0]["stacktrace"]["frames"][-1]["vars"]
 
     if integrations:
-        assert sorted(frame_vars.keys()) == ["foo", "foo.d", "foo.d[1]"]
-        assert frame_vars["foo.d"] == {"1": "2"}
-        assert frame_vars["foo.d[1]"] == "2"
+        # Values closest to the exception line appear first
+        # Test this order if possible given the Python version and dict order
+        expected_keys = [
+            "namespace",
+            "namespace.d",
+            "namespace.d[1]",
+            "s",
+            "a",
+            "p",
+            "o",
+            "i",
+            "u",
+            "y",
+        ]
+        if sys.version_info[:2] == (3, 5):
```

Comment:
Instead of this conditional can't we do `assert set(frame_vars.keys()) == set(expected_keys)`

Reply:
I think you misunderstood. `frame_vars.keys()` is set-like and so `assert frame_vars.keys() == set(expected_keys)` would work in all versions. `assert list(frame_vars.keys()) == expected_keys` is a stronger assertion to make extra sure (when possible) that values are prioritised as expected. The tests are fine as they are, the reason for the failure in 3.5 is that the implementation actually doesn't work and the values are actually wrong.

The problem is that I `return OrderedDict` above so that the trimming in `serialize` will keep the first 10 items of the dict, but just before the trimming the `OrderedDict` gets converted to a `dict` (see the PR description) which breaks the order in 3.5 and thus the implementation (the wrong values get trimmed). I'm asking if I should just trim the values within the pure_eval integration and forget the `OrderedDict`, or try to preserve order in `serialize`.

Reply:
To be clear both options should be equivalent right now, but if `serialize` changes its algorithm and starts trimming things more cleverly in the future, it will be better if it has the full dict of variables in order of priority rather than a dict that was eagerly/preemptively trimmed by pure_eval.

Reply:
Ah right, I recommend going for option 3 but avoiding hardcoding the number 10 (instead importing the var). I don't think there's a safe alternative to `dict` outside of pure_eval context, so I'd like to avoid changes to serializer.

Reply:
>but if serialize changes its algorithm and starts trimming things more cleverly in the future

I doubt this will happen in serialization, let's assume pure_eval always knows better (seems likely given the amount of extra analysis it does)

Reply:
Done